CAM: Test `PathTests.TestPathDrillable.TestPathDrillable` fails on s390x

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=15&t=71764

### Version

0.20 (Release)

### Full version info

```shell
Debian packaged version 0.20.1 -- Debian version 0.20.1+dfsg1-1

This information is from my amd64 machine, not from s390x, but versions should be the same.
]
OS: Debian GNU/Linux bookworm/sid 
Word size of FreeCAD: 64-bit
Version: 0.20.1.
Build type: Release
Python 3.10.6, Qt 5.15.4, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: English/United Kingdom (en_GB)
```


### Subproject(s) affected?

Path

### Issue description

Hi,

I'm one of the people looking after FreeCAD in Debian, I've encountered a test suite error in the TestPathDrillable testsuite on the s390x Architecture.
The other architectures seems to be fine, so I'm quite puzzled about the reason behind.

s390x is for Debian a release architecture, so this failure currently prevents freecad to migrate to "testing", the staging area for the next release of Debian.

The failure happens on the Debian CI, the full log for (the failing) s390x can be found here: [https://ci.debian.net/data/autopkgtest/ ... 237/log.gz](https://ci.debian.net/data/autopkgtest/testing/s390x/f/freecad/25840237/log.gz), here the excerpt of the failing test:

```
======================================================================
FAIL: test10 (PathTests.TestPathDrillable.TestPathDrillable)
Test isDrillable
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/freecad/Mod/Path/PathTests/TestPathDrillable.py", line 78, in test10
    self.assertTrue(drillableLib.isDrillable(self.obj.Shape, candidate))
AssertionError: False is not true

======================================================================
FAIL: test20 (PathTests.TestPathDrillable.TestPathDrillable)
Test getDrillableTargets
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/freecad/Mod/Path/PathTests/TestPathDrillable.py", line 305, in test20
    self.assertEqual(len(results), 20)
AssertionError: 21 != 20

----------------------------------------------------------------------
Ran 736 tests in 46.072s

FAILED (failures=2)
```

As said, I have no idea where to start looking to pin point the problem, let me know if I can somehow provide more information.

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

@luzpaz  thanks for the ping, I acutally forgot to follow up…

s390x situation is unfortunately worse now, as there are no [Qt5WebEngineWidgets for s390x](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=913043), so I had to configure freecad with ` -DBUILD_WEB=OFF -DBUILD_START=OFF` to get it compiling.

FTR, I was compiling 381f534c477b08148f956ac47e8341e63d3ac432 on the s390x porterbox.


```
======================================================================
FAIL: test10 (PathTests.TestPathDrillable.TestPathDrillable.test10)
Test isDrillable
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobi/FreeCAD/build/Mod/Path/PathTests/TestPathDrillable.py", line 77, in test10
    self.assertTrue(Drillable.isDrillable(self.obj.Shape, candidate))
AssertionError: False is not true

======================================================================
FAIL: test20 (PathTests.TestPathDrillable.TestPathDrillable.test20)
Test getDrillableTargets
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobi/FreeCAD/build/Mod/Path/PathTests/TestPathDrillable.py", line 272, in test20
    self.assertEqual(len(results), 20)
AssertionError: 21 != 20

----------------------------------------------------------------------
Ran 1098 tests in 56.051s

FAILED (failures=2)

```

--- a/src/Mod/CAM/Path/Base/Drillable.py
+++ b/src/Mod/CAM/Path/Base/Drillable.py
@@ -244,10 +244,10 @@ def compareVecs(vec1, vec2, exact=False):
     angle = 0 if math.isnan(angle) else math.degrees(angle)
     Path.Log.debug("vector angle: {}".format(angle))
     if exact:
-        return numpy.isclose(angle, 0, rtol=1e-05, atol=1e-06)
+        return numpy.isclose(angle, 0, rtol=1e-05, atol=1e-04)
     else:
-        return numpy.isclose(angle, 0, rtol=1e-05, atol=1e-06) or numpy.isclose(
-            angle, 180, rtol=1e-05, atol=1e-06
+        return numpy.isclose(angle, 0, rtol=1e-05, atol=1e-04) or numpy.isclose(
+            angle, 180, rtol=1e-05, atol=1e-04
         )
 
 

